core/src/com/marama/view/util/MaramaficationLoader.java

```diff
@@ -0,0 +1,69 @@
+package com.marama.view.util;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.assets.AssetDescriptor;
+import com.badlogic.gdx.assets.AssetLoaderParameters;
+import com.badlogic.gdx.assets.AssetManager;
+import com.badlogic.gdx.assets.loaders.AsynchronousAssetLoader;
+import com.badlogic.gdx.assets.loaders.FileHandleResolver;
+import com.badlogic.gdx.files.FileHandle;
+import com.badlogic.gdx.utils.Array;
+import com.badlogic.gdx.utils.JsonReader;
+import com.badlogic.gdx.utils.JsonValue;
+import com.marama.view.entities.Maramafication;
+
+import java.util.UUID;
+
+/**
+ * Loads and stores maramafications.
+ */
+public class MaramaficationLoader extends AsynchronousAssetLoader<Maramafication, MaramaficationLoader.MaramaficationLoaderParameter> {
+    // The currently loaded maramafication.
+    private Maramafication maramafication;
+    // Remember the JsonReader so that it can be used for all maramafications
+    private JsonReader jsonReader;
+
+    public MaramaficationLoader(FileHandleResolver resolver) {
+        super(resolver);
+        jsonReader = new JsonReader();
+    }
+
+    @Override
+    public void loadAsync(AssetManager manager, String fileName, FileHandle file, MaramaficationLoader.MaramaficationLoaderParameter parameter) {
+        // First make sure the current is cleared
+        maramafication = null;
+
+        // Since the maramafications are currently saved as json files, read the json file
+        // This should be altered when there is another way of passing maramafication information
```

Comment:
Backlog card if you think this is something that ought to be done.

Reply:
This'll automatically be done if we can communicate with scala. I can create a backlog item, though.